test/addons-napi/test_reference/test.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+// Flags: --expose-gc
+
+const common = require('../../common');
+const assert = require('assert');
+
+const test_reference = require(`./build/${common.buildType}/test_reference`);
+
+// This test script uses external values with finalizer callbacks
+// in order to track when values get garbage-collected. Each invocation
+// of a finalizer callback increments the finalizeCount property.
+assert.strictEqual(0, test_reference.finalizeCount);
+
+// External value without a finalizer
+let value = test_reference.createExternal();
```

Comment:
`const` won't work because `value` must later be set to `null` in order to allow it to be garbage-collected.

... unless the scope of the `{` `}` blocks is narrowed to just around where each `value` is live. But then I think that would actually hurt readability because the blocks would not correspond to the logical test cases.